src/auth/src/credentials.rs

```diff
@@ -253,13 +256,20 @@ pub async fn create_access_token_credential() -> Result<Credential> {
     }
 }
 
+#[derive(Debug, PartialEq)]
+enum AdcPathKind {
+    FromEnv,
+    WellKnown,
+}
+
```

Comment:
I think you want something like:

```suggestion
#[derive(Debug, PartialEq)]
enum AdcPath {
    FromEnv(String),
    WellKnown(String)
}
```

and then return `fn adc_path() -> Option<AdcPath>` or you want:

```suggestion
#[derive(Debug, PartialEq)]
enum AdcPath {
    FromEnv(String),
    WellKnown(String),
    None
}
```

and then `adc_path() -> AdcPathKind`